engine/src/main/java/org/terasology/engine/subsystem/rpc/DiscordRPCSubSystem.java

```diff
@@ -0,0 +1,308 @@
+    private int reconnectTries = 1;
+    private boolean connectedBefore;
+    private int lastPing;
+    private Config config;
+    private String lastState;
+    private boolean dontTryAgain;
+    private boolean enabled;
+
+    public DiscordRPCSubSystem() throws IllegalStateException {
+        if (instance != null) {
+            throw new IllegalStateException("More then one instance in the DiscordRPC");
+        }
+        lastRichPresence = null;
+        ipcClient = new IPCClient(CLIENT_ID);
+        ipcClient.setListener(this);
+        autoReconnect = true;
+        reconnectThread = new Thread(this);
+        reconnectThread.setName("DISCORD-RPC-RECONNECT");
+        reconnectThread.start();
+        instance = this;
+        enabled = false;
+        dontTryAgain = true;
+    }
+
+    public void sendRichPresence(RichPresence richPresence) {
+        this.lastRichPresence = richPresence;
+        if (!ready || lastRichPresence == null || !enabled) {
+            return;
+        }
+        ipcClient.sendRichPresence(lastRichPresence);
+    }
+
+    @Override
+    public void onReady(IPCClient client) {
+        if (reconnecting) {
+            getLogger().info("Discord RPC >> Reconnected!");
+            reconnectTries = 1;
+        } else {
+            getLogger().info("Discord RPC >> Connected!");
+            connectedBefore = true;
+        }
+        this.ipcClient = client;
+        if (!ready) {
+            ready = true;
+        }
+        if (lastRichPresence == null) {
+            RichPresence.Builder builder = new RichPresence.Builder();
+            builder.setLargeImage(LARGE_IMAGE);
+            lastRichPresence = builder.build();
+        }
+        client.sendRichPresence(lastRichPresence);
+    }
+
+    @Override
+    public void onDisconnect(IPCClient client, Throwable t) {
+        if (ready) {
+            ready = false;
+        }
+        getLogger().info("Discord RPC >> Disconnected!");
+    }
+
+    @Override
+    public void run() {
+        while (autoReconnect) {
+            try {
+                // Ignore if the Discord RPC is not enabled
+                if (!enabled) {
+                    if (ready) {
+                        try {
+                            getInstance().ipcClient.close();
+                        } catch (Exception ex) { }
+                    }
+                    Thread.sleep(1);
+                    continue;
+                }
+
+                // Don't try to retry to discover the RPC until some reasons happen
+                if (dontTryAgain) {
+                    Thread.sleep(1);
+                    continue;
+                }
+
+                // Connect if the connect on init didn't connect successfully
+                if (!connectedBefore && !ready) {
+                    lastPing = 0;
+                    try {
+                        ipcClient.connect();
+                    } catch (Exception ex) { }
+                    Thread.sleep(15 * 1000);
```

Comment:
You've got a few "magic numbers" in this area involved in your math. Those are generally better as constants named something informative. Like instead of `reconnectTries >= 5` add a constant `RECONNECT_TRIES = 5`

Reply:
**Done** I replaced all the constant stuff to the constant variable. 😄